### ISSUE TITLE
VTOL transition switch broken in current master

### ISSUE THREAD

**Issue by sanderux** [PX4/Firmware#4466]
It appears to be related to the radio settings
Transition switch is ignored (quad spins up with throttle when switch on)
Also, the pusher ESC does not initialize, or only does so after changing the VT_IDLE_PWM_MC, it does not seem to matter to what direction to change this.

@AndreasAntener @tumbili @LorenzMeier

**Comment by LorenzMeier** [comment]
Make sure to run "make clean" and compile master fresh before uploading

**Comment by sanderux** [comment]
Was done, also been able to reproduce om master precompiled through QGC
Additionally the wing servo's do not respond so i'm quite sure this
https://github.com/PX4/Firmware/blob/master/src/modules/vtol_att_control/vtol_att_control_main.cpp#L466
returns false. When reverting to older firmware (28 march) the problem disappears

**Comment by LorenzMeier** [comment]
Anyone feeling like adding a proper transition switch? We could squeeze that in for 1.3 and also add QGC config support. Would be nice to have this less obscure.

**Comment by sanderux** [comment]
Last working commit i was able to compile: https://github.com/PX4/Firmware/commit/ebacd4c1de5a0231ad9a4cddcc3759a5389835ea

First failure commit i was able to compile:
https://github.com/PX4/Firmware/commit/4bdfb8a10775ab2f8feeade1c622de259234052c

**Comment by sanderux** [comment]
It seems to have broken on Apr 11 after @kd0aij's commit. I run into a lot of compile errors when attempting to compile half-way. 
@LorenzMeier can you have a look?

**Comment by tubeme** [comment]
We found out this just yesterday! Making a new machine with new Pix flashing Master, loading our perfect settings set and DISASTER. On the bench test, no transition. Also programming acro switch has a very strange behavior in the QGC Flight Mode Setup Page. When we switch to stabilize the QGC reports it in the status bar the voice pronounce it correct but in the Flight Mode setup page it show we are in Acro and vice versa.

Also we could not take off with the quad. The quad seemed like it is in total disorder, even worse than acro. The VTOL just flipped and flipped and flipped... no take off. We had to move back to Stable... But it lacks a lot of the new things.
